### PR TITLE
done

### PR DESCRIPTION
**Pull request by siarotchi**

### CODE REVIEW

**Comment by wXrey** [comment]
Test comment

**Comment by siarotchi** [comment]
123

File: index.js [hunk 1]
```diff
@@ -1,26 +1,118 @@
...
+fibo(6);
+fibo(7);
 
 // 2. Implementing with reduce
 
 //concat;
+let arr = [1, 2];
+let arr1 = [3, 4];
+
+const concatFn = (array, newEl) =>
```
Comment (MihailGorceagEN) [inline index.js:16]: you are mutating state, please note that `let arr = [1, 2];` is changed, please don't change the variable

File: index.js [hunk 2]
```diff
@@ -1,26 +1,118 @@
...
+    name: "Michael",
+    age: 25,
+  },
+  {
+    name: "Lisa",
+    age: 30,
+  },
+];
+
+let adult = friends.reduce(function (newArr, person) {
```
Comment (MihailGorceagEN) [inline index.js:38]: please don't mutate state, use spread instead of push

File: index.js [hunk 3]
```diff
@@ -1,26 +1,118 @@
...
+  }
+  return acc;
+}, null);
+
 // includes;
+let arr = [1, 3, 4, 5];
+
+let includesTwo = arr.reduce((result, item) => {
+  if (item === 2) {
+    result = true;
```
Comment (MihailGorceagEN) [inline index.js:84]: please don't mutate state, `return true;`

File: index.js [hunk 4]
```diff
@@ -1,26 +1,118 @@
...
 // reverse;
+let arr = [1, 2, 3, 4];
+console.log(arr.reduce((a, b) => [b, ...a], []));
+
 // some;
+let arr = [-1, -1, 2, -2, -3];
+
+let somePositive = arr.reduce((result, item) => {
+  if (item > 0) {
+    result = true;
```
Comment (MihailGorceagEN) [inline index.js:109]: please don't assign value to result

File: index.js (hunk 2 above)
Comment (siarotchi) [reply index.js:38]: but here is a new array -- i didnt mutate it

**Pull request merged**